docs/index.rst

```diff
@@ -152,6 +156,18 @@ There are some additional tools, mostly of use for maintainers, in the
 `astropy/astropy-procedures repository
 <https://github.com/astropy/astropy-procedures>`__.
 
+{%else%}
+
+To read the developer documentation, you will need to go to the
+`latest developer version of the documentation <https://docs.astropy.org/en/latest/#developer-documentation>`__.
```

Comment:
I wonder if it wouldn't make sense to move the entire developer docs TOC to its own page, perhaps with a bit of expansion.  If you'd like, I can make a quick prototype of what I have in mind in a separate PR, or added to this one.  I am planning to do some overhaul of the developer docs anyways so it would be a good opportunity to get my feet wet.

Reply:
This PR is already merged last week, so you would need to open a follow-up PR. 😉

Reply:
Oh, somehow I didn't realize this was already merged.  Ok then, if you think it's a good idea I can make a follow-up.

Reply:
Yeah, feel free to PR. The dev docs definitely need some TLC.